src/sonic-config-engine/sonic-cfggen

```diff
@@ -327,14 +343,13 @@ def main():
             hardware_data['DEVICE_METADATA']['localhost'].update(asic_id=asic_id)
         deep_update(data, hardware_data)
 
-    if args.template is not None:
-        template_file = os.path.abspath(args.template)
-        paths  = ['/', '/usr/share/sonic/templates', os.path.dirname(template_file)]
-        if args.template_dir is not None:
-            template_dir = os.path.abspath(args.template_dir)
-            paths.append(template_dir)
-        loader = jinja2.FileSystemLoader(paths)
+    paths = ['/', '/usr/share/sonic/templates']
+    if args.template != None:
+        args.batch.append((args.template, sys.stdout))
```

Comment:
>args.batch [](start = 8, length = 10)

We need to treat args as readonly

Reply:
why are we imposing this?

Reply:
args is the return value of `parser.parse_args()`. You have the freedom to reuse it in order to save one temporary variable, but this bring side-effect and is weird to a normal code reader.

---
In reply to: [453049703](https://github.com/Azure/sonic-buildimage/pull/4941#discussion_r453049703) [](ancestors = 453049703)